### PR TITLE
Task-37845 : Active DLP document preview

### PR DESCRIPTION
**Pull request by ahmadmido**

### CODE REVIEW

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItem.java [hunk 1]
```diff
@@ -34,6 +38,14 @@ public void setTitle(String title) {
         this.title = title;
     }
 
+    public String getRefernece() { return refernece; }
+
+    public void setRefernece(String refernece) { this.refernece = refernece; }
```
Comment (plamarque) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItem.java]:
```suggestion
    public void setReference(String reference) { this.reference = reference; }
```

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItem.java [hunk 2]
```diff
@@ -34,6 +38,14 @@ public void setTitle(String title) {
         this.title = title;
     }
 
+    public String getRefernece() { return refernece; }
```
Comment (plamarque) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItem.java]:
```suggestion
    public String getReference() { return reference; }
```

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItem.java [hunk 3]
```diff
@@ -12,6 +12,10 @@
 
     private String keywords;
 
+    private String refernece;
```
Comment (plamarque) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/dto/DlpPositiveItem.java]:
```suggestion
    private String reference;
```

**Review by plamarque — DISMISSED** [review]
typo on "reference"

**Review by rdenarie — DISMISSED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/domain/DlpPositiveItemEntity.java [hunk 4]
```diff
@@ -42,6 +42,9 @@
     @Column(name = "KEYWORDS")
     private String keywords;
 
+    @Column(name = "ITEM_DOWNLOAD_URL")
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/domain/DlpPositiveItemEntity.java]: In fact, I think we should not store that, for 2 reasons :
1) If DLPPositiveItem is related to an activity, or a wiki or something different from a document, we will not have a "download" url
2) This url can be computed at display with the item reference. If the document is moved in Security folder from the moment it is put in quarantine and the moment it is previewed, the url can change. By computing it at display, we ensure that the url is still ok

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java [hunk 5]
```diff
@@ -52,6 +55,29 @@ public Response getDlpPositiveItems(@ApiParam(value = "Offset", required = false
         }
     }
 
+    @GET
+    @Path("/itemUrl/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed("administrators")
+    @ApiOperation(value = "Retrieves the url of dlp positive items", httpMethod = "GET", response = Response.class, produces = "application/json",
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpItemRestServices.java]: Why do we need a specific rest service only for url ?
Is it not possible to get it when getting the item list ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**